cmd/erasure-server-pool.go

```diff
@@ -2029,7 +2029,10 @@ func (z *erasureServerPools) Walk(ctx context.Context, bucket, prefix string, re
 								return
 							}
 
-							versionsSorter(fivs.Versions).reverse()
+							// Note: entry.fileInfoVersions returns versions sorted in reverse chronological order based on ModTime
+							if opts.WalkVersionsSort == WalkVersionsSortAsc {
```

Comment:
I just realized that this PR introduces support for receiving versions of an object in reverse chronological order of `ModTime`s. To minimize the amount of files that need to be touched, I use the 'zero' value of [`WalkVersionsSortOrder`](https://github.com/minio/minio/pull/17946/files#diff-7410eec4482a2c3834d74dd077538d3b377627a3a50e436a372bdd78934ef837R49) to indicate ascending order as default ordering. I am highlighting this to draw review attention to this, since it affects more than batch-expiry.